pyomo/contrib/alternative_solutions/aos_utils.py

```diff
@@ -0,0 +1,290 @@
+        )
+        _filter_model_variables(
+            variable_set,
+            var_generator,
+            include_continuous,
+            include_binary,
+            include_integer,
+            include_fixed,
+        )
+    else:
+        for comp in components:
+            if hasattr(comp, "ctype") and comp.ctype == pe.Block:
+                blocks = comp.values() if comp.is_indexed() else (comp,)
+                for item in blocks:
+                    variables = vfe.get_vars_from_components(
+                        item, component_list, include_fixed=include_fixed
+                    )
+                    _filter_model_variables(
+                        variable_set,
+                        variables,
+                        include_continuous,
+                        include_binary,
+                        include_integer,
+                        include_fixed,
+                    )
+            elif (
+                isinstance(comp, tuple)
+                and hasattr(comp[0], "ctype")
+                and comp[0].ctype == pe.Block
+            ):
+                block = comp[0]
+                descend_into = pe.Block if comp[1] else False
+                blocks = block.values() if block.is_indexed() else (block,)
+                for item in blocks:
+                    variables = vfe.get_vars_from_components(
+                        item,
+                        component_list,
+                        include_fixed=include_fixed,
+                        descend_into=descend_into,
+                    )
+                    _filter_model_variables(
+                        variable_set,
+                        variables,
+                        include_continuous,
+                        include_binary,
+                        include_integer,
+                        include_fixed,
+                    )
+            elif hasattr(comp, "ctype") and comp.ctype in component_list:
+                constraints = comp.values() if comp.is_indexed() else (comp,)
+                for item in constraints:
+                    variables = pe.expr.identify_variables(
+                        item.expr, include_fixed=include_fixed
+                    )
+                    _filter_model_variables(
+                        variable_set,
+                        variables,
+                        include_continuous,
+                        include_binary,
+                        include_integer,
+                        include_fixed,
+                    )
+            elif hasattr(comp, "ctype") and comp.ctype == pe.Var:
+                variables = comp.values() if comp.is_indexed() else (comp,)
+                _filter_model_variables(
+                    variable_set,
+                    variables,
+                    include_continuous,
+                    include_binary,
+                    include_integer,
+                    include_fixed,
+                )
```

Comment:
I'm not sure.  @jlgearh ???

Reply:
We discussed this and this is here because it is allowed to pass in Vars to this method and this acts as a sort of flattener. This code could be simplified, but after this PR.